presto-main/src/test/java/com/facebook/presto/operator/TestDriver.java

```diff
@@ -116,6 +117,25 @@ public void testNormalFinish()
         assertTrue(source.isFinished());
     }
 
+    @Test(invocationCount = 10_000, timeOut = 5_000)
```

Comment:
This test can repro the issue with 10K iterations and it takes ~2m on my laptop, so I am not sure whether we want to check this in. I just put this up to show how this can be reproduced. If you also think that we shouldn't merge this test, I can remove it.